lib/code_corps/github/api.ex

```diff
@@ -0,0 +1,28 @@
+defmodule CodeCorps.Github.API do
+  @client_secret Application.get_env(:code_corps, :github_client_secret)
+  @client_id Application.get_env(:code_corps, :github_client_id)
+
+  @base_connect_params %{
+    client_id: @client_id,
+    client_secret: @client_secret
+  }
+
+  def connect(code) do
+    with {:ok, %HTTPoison.Response{body: response}} <- code |> build_connect_params() |> do_connect(),
+      {:ok, %{"access_token" => access_token}} <- response |> Poison.decode
+    do
+      {:ok, access_token}
+    else
+      {:ok, %{"error" => error}} -> {:error, error}
```

Comment:
It is what it is, though. Seems that github itself returns a 200 ok, with an error hash. Not much we can do about it, other than interpret it here.